### ISSUE TITLE
Place the 'web' and 'grafana' pods in appropriate `serviceAccountNames`s

### ISSUE THREAD

**Issue by codeman9** [linkerd/linkerd2#1977]
### What problem are you trying to solve?
The `web` pods and the `grafana` pods are not currently linked to a service account and my PSP rules are applied to the various linkerd service accounts. This means that these pods are using the `default` service account, which is generally locked down in my company.

### How should the problem be solved?
Add `serviceAccountName: linkerd-controller` to the `web` pod
Add `serviceAccountName: linkerd-prometheus` to the `grafana` pod

Also, if appropriate change existing usage of `serviceAccount` to `serviceAccountName` since `serviceAccount` has been deprecated.

From the K8s 1.10 api reference:
```
DeprecatedServiceAccount is a depreciated alias for ServiceAccountName. Deprecated: Use serviceAccountName instead.
```

### Any alternatives you've considered?
I considered making more service accounts for these, but I think it makes sense to use what is already provided.

### How would users interact with this feature?
No action needed by the users.

**Comment by codeman9** [comment]
@klingerf This is pretty straightforward. Just have a question about whether you agree that the `grafana` deployment should be linked to the `linkerd-prometheus` service account or another one.

**Comment by klingerf** [comment]
👍 Thanks for creating this issue!

Switching from `serviceAccount` to `serviceAccountName` in our install configs would be great.

I'm inclined to make two new service accounts for the two deployments that are currently running with the default service account, since those deployments do not need any of the RBAC permissions that are granted to the existing service accounts. And if they eventually do need specific RBAC permissions, I think we'll be better off running each deployment under a different service account. Something like:

```
### Service Account Web ###
---
kind: ServiceAccount
apiVersion: v1
metadata:
  name: linkerd-web
  namespace: linkerd

### Service Account Grafana ###
---
kind: ServiceAccount
apiVersion: v1
metadata:
  name: linkerd-grafana
  namespace: linkerd
```

@codeman9 Does that work for you?

**Comment by codeman9** [comment]
@klingerf Yep! Works for me.

**Comment by klingerf** [comment]
Fixed by #1978.
